pkg/agent/controller/traceflow/packetin.go

```diff
@@ -89,17 +90,28 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*opsv1alpha1.Tracefl
 		return nil, nil, fmt.Errorf("unsupported traceflow packet Ethertype: %d", pktIn.Data.Ethertype)
 	}
 
-	// Get traceflow CRD from cache by data plane tag.
-	tf, err := c.GetRunningTraceflowCRD(uint8(tag))
+	firstPacket := false
+	c.runningTraceflowsMutex.RLock()
+	tfState, exists := c.runningTraceflows[tag]
+	if exists {
+		firstPacket = !tfState.receivedPacket
+		tfState.receivedPacket = true
+	}
+	c.runningTraceflowsMutex.RUnlock()
+
+	if firstPacket {
+		// Uinstall the OVS flows after receiving the first packet.
+		c.ofClient.UninstallTraceflowFlows(tag)
```

Comment:
Yes, that is true.
I am not sure it is theoretically possible that the flow deletion can be realized before all the packets from the same pipeline are sent, but even that is possible at least it is not quite possible in reality right?

Reply:
I think it is highly unlikely (if the packet was not recirculated in the datapath, I think it would be impossible, but I'm no OVS expert), so we can probably keep your current code but add a comment.

Reply:
As implementation in this patch, I think the flow deletion will be always after packets processed in unicast environment, but this may not correct in multicast if we want to support it in future.
Meanwhile, sometimes Traceflow can capture unexpected network issues, such as duplicate packet, thus we may not need to delete the flows immediately. A hard timeout is enough.

Reply:
@gran-vmv : I changed to delete flows mainly for live-traffic case to avoid many packets of different connections being captured. But I got your point about packet dup, etc., so I changed the code to delete flows only for live-traffic Traceflow.

Reply:
Sounds good.